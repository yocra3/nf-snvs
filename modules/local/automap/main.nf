process AUTOMAP {
    tag "$meta.id"
    label 'process_single'
    errorStrategy 'retry'

    // Conda is not supported
    container "/mnt/tblab/yolanda/automap/automap.sif"
    
    input:
    tuple val(meta), path(vcf), path(tbi)
    val automap_assembly
    path projectDir

    output:
    tuple val(meta), path("*HomRegions*"), emit: roh_automap
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    
    if ( task.attempt == 1 )
        """
        if [[ \$(bcftools query -l ${vcf} | wc -l) -gt 1 ]]; then
		    for sample in \$(bcftools query -l ${vcf}); do

			    bcftools view -s \${sample} -O v -o \${sample}.indv.vcf ${vcf}
					
			    bash /app/AutoMap_v1.3.sh \\
			    --vcf \${sample}.indv.vcf \\
			    --out . \\
			    --genome ${automap_assembly}

			    mv \${sample}/*HomRegions* .

		    done
			
	    else
					
		    bcftools view -O v -o ${vcf}.vcf ${vcf}

		    bash /app/AutoMap_v1.3.sh \\
		    --vcf ${vcf}.vcf \\
		    --out . \\
		    --genome ${automap_assembly}

		    mv \$(bcftools query -l ${vcf})/*HomRegions* .

	    fi
        
        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            bcftools: \$(bcftools --version 2>&1 | head -n1 | sed 's/^.*bcftools //; s/ .*\$//')
        END_VERSIONS
	    """
    
    else
        """
        echo "Less than 10k variants (with quality)" > ${prefix}_no_automap_HomRegions.txt
		
        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            bcftools: \$(bcftools --version 2>&1 | head -n1 | sed 's/^.*bcftools //; s/ .*\$//')
        END_VERSIONS
        """
}
