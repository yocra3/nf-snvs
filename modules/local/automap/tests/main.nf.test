// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test automap
nextflow_process {

    name "Test Process AUTOMAP"
    script "../main.nf"
    process "AUTOMAP"

    tag "modules"
    tag "modules_nfcore"
    tag "automap"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("automap - vcf") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("/mnt/tblab/GURYBQ/10-1918.hg38.final.vcf.gz", checkIfExists: true),
                    file("/mnt/tblab/GURYBQ/10-1918.hg38.final.vcf.gz.tbi", checkIfExists: true)
                ]
                input[1] = "hg38"
                input[2] = "${projectDir}"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

}
