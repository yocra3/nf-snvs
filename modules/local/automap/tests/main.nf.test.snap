{
    "automap - vcf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "10-1918.HomRegions.pdf:md5,246fd868eab01114d31420c992d1cc24",
                            "10-1918.HomRegions.tsv:md5,48928f23c6ff528a7ef5cf9ea9d3c7ae"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,db647745175c1b36911a7e1290d82a74"
                ],
                "roh_automap": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "10-1918.HomRegions.pdf:md5,246fd868eab01114d31420c992d1cc24",
                            "10-1918.HomRegions.tsv:md5,48928f23c6ff528a7ef5cf9ea9d3c7ae"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,db647745175c1b36911a7e1290d82a74"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-12T13:23:08.87226653"
    }
}